habitat/utils/gym_adapter.py

```diff
@@ -49,6 +51,96 @@ def transform_shape(shape):
     )
 
 
+def _is_continuous(original_space: gym.Space) -> bool:
+    """
+    returns true if the original space is only suitable for continuous control
+    """
+    if isinstance(original_space, spaces.Box):
+        # Any Box means it is continuous
+        return True
+    if isinstance(original_space, EmptySpace):
+        return False
+    if isinstance(original_space, Mapping):
+        return any((_is_continuous(v) for v in original_space.values()))
+    raise NotImplementedError(
+        f"Unknow action space found : {original_space}. Can only be Box or Empty"
+    )
+
+
+def _recursive_continuous_size_getter(
+    original_space: gym.Space, size: List[int]
+):
+    """
+    Returns the size of a continuous action vector from a habitat environment action space
+    """
+    if isinstance(original_space, spaces.Box):
+        size[0] += original_space.shape[0]
+    elif isinstance(original_space, EmptySpace):
+        size[0] += 1
+    elif isinstance(original_space, Mapping):
+        for v in original_space.values():
+            _recursive_continuous_size_getter(v, size)
+    else:
+        raise NotImplementedError(
+            f"Unknow continuous action space found : {original_space}. Can only be Box, Empty or Dict."
+        )
+
+
+def create_action_space(original_space: gym.Space) -> gym.Space:
+    """
+    Converts a habitat task action space into a either continuous (Box) or discrete (Discrete) gym.space.
+    """
+    assert isinstance(
+        original_space, Mapping
+    ), f"The action space of the environment needs to be a Mapping, but was {original_space}"
+    if _is_continuous(original_space):
+        size = [0]
+        _recursive_continuous_size_getter(original_space, size)
+        return spaces.Box(
+            shape=(size[0],), low=-1.0, high=1.0, dtype=np.float32
```

Comment:
Since we are doing the check that the action is in bounds, I think it is important that we copy the low and high from the sub-action spaces. It should not always be -1,1.

Reply:
It was [-1,1] in the original code. This was meant to be a bug fix. I will try an implement it.